question-servers/freeformPythonLib/prairielearn.py

```diff
@@ -776,8 +772,8 @@ def string_to_2darray(s, allow_complex=True):
         if (m == 0):
             return (None, {'format_error': 'Matrix has no rows.'})
 
-        # Get number of columns by splitting first row on space
-        n = len(s[0].split())
+        # Get number of columns by splitting first row on space (treat comma as space)
+        n = len(s[0].replace(',', ' ').split())
```

Comment:
I'm thinking now that the `replace`s can be refactored to lines 766 and 938 (but haven't tested).

Reply:
Done+tested on my end